<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget parent="main_window" name="Main Window">
  <Tab tab_name="tab1" containers="1">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.105479" right="33.646000" bottom="0.042334"/>
       <limitY/>
       <curve color="#ff7f0e" name="/iq_model_calib_rt/iq_cal_node/K_d0_cal.0"/>
       <curve color="#f14cc1" name="/iq_model_calib_rt/iq_cal_node/K_d0_cal.1"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.211924" right="33.646000" bottom="-10.593971"/>
       <limitY/>
       <curve color="#9467bd" name="/iq_model_calib_rt/iq_cal_node/K_d1_cal.0"/>
       <curve color="#17becf" name="/iq_model_calib_rt/iq_cal_node/K_d1_cal.1"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="tab2" containers="1">
   <Container>
    <DockSplitter count="1" orientation="-" sizes="1">
     <DockSplitter count="2" orientation="|" sizes="0.5;0.5">
      <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range left="13.650000" top="0.043876" right="33.646000" bottom="0.043836"/>
         <limitY/>
         <curve color="#ff7f0e" name="/iq_model_calib_rt/iq_cal_node/K_d0_cal.0"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range left="13.650000" top="0.103978" right="33.646000" bottom="0.103864"/>
         <limitY/>
         <curve color="#f14cc1" name="/iq_model_calib_rt/iq_cal_node/K_d0_cal.1"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range left="13.650000" top="-0.045092" right="33.646000" bottom="-0.056344"/>
         <limitY/>
         <curve color="#9467bd" name="/iq_model_calib_rt/iq_cal_node/K_d1_cal.0"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
         <range left="13.650000" top="-10.302943" right="33.646000" bottom="-10.337511"/>
         <limitY/>
         <curve color="#17becf" name="/iq_model_calib_rt/iq_cal_node/K_d1_cal.1"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="tab3" containers="1">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="1.025000" right="33.646000" bottom="-0.024996"/>
       <limitY/>
       <curve color="#bcbd22" name="/iq_model_calib_rt/iq_cal_node/alpha_f0.0"/>
       <curve color="#d62728" name="/iq_model_calib_rt/iq_cal_node/alpha_f1.0"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="1.024996" right="33.646000" bottom="-0.024833"/>
       <limitY/>
       <curve color="#1f77b4" name="/iq_model_calib_rt/iq_cal_node/alpha_f0.1"/>
       <curve color="#1ac938" name="/iq_model_calib_rt/iq_cal_node/alpha_f1.1"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="tab8" containers="1">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.000398" right="33.646000" bottom="-0.000461"/>
       <limitY/>
       <curve color="#ff7f0e" name="/iq_model_calib_rt/iq_est_node/iq_est.0"/>
       <curve color="#9467bd" name="/iq_model_calib_rt/iq_cal_node/iq_meas.0"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.011342" right="33.646000" bottom="-0.011220"/>
       <limitY/>
       <curve color="#f14cc1" name="/iq_model_calib_rt/iq_est_node/iq_est.1"/>
       <curve color="#17becf" name="/iq_model_calib_rt/iq_cal_node/iq_meas.1"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="tab7" containers="1">
   <Container>
    <DockSplitter count="1" orientation="-" sizes="1">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="4.100000" right="33.646000" bottom="-0.100000"/>
       <limitY/>
       <curve color="#1f77b4" name="/iq_model_calib_rt/iq_cal_node/sol_millis"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="tab4" containers="1">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.044891" right="33.646000" bottom="0.042939"/>
       <limitY/>
       <curve color="#ff7f0e" name="/iq_model_calib_rt/iq_cal_node/tau_friction_meas.0"/>
       <curve color="#1f77b4" name="/iq_model_calib_rt/iq_est_node/tau_friction.0"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.113966" right="33.646000" bottom="0.050903"/>
       <limitY/>
       <curve color="#f14cc1" name="/iq_model_calib_rt/iq_cal_node/tau_friction_meas.1"/>
       <curve color="#d62728" name="/iq_model_calib_rt/iq_est_node/tau_friction.1"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="tab5" containers="1">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.001675" right="33.646000" bottom="-0.045482"/>
       <limitY/>
       <curve color="#9467bd" name="/iq_model_calib_rt/iq_cal_node/tau_total.0"/>
       <curve color="#bcbd22" name="/iq_model_calib_rt/iq_cal_node/tau_linkside_filt.0"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.023489" right="33.646000" bottom="-0.094780"/>
       <limitY/>
       <curve color="#17becf" name="/iq_model_calib_rt/iq_cal_node/tau_total.1"/>
       <curve color="#1f77b4" name="/iq_model_calib_rt/iq_cal_node/tau_linkside_filt.1"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="tab9" containers="1">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.044891" right="33.646000" bottom="0.042939"/>
       <limitY/>
       <curve color="#ff7f0e" name="/iq_model_calib_rt/iq_cal_node/tau_friction_meas.0"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
       <range left="13.650000" top="0.113966" right="33.646000" bottom="0.050903"/>
       <limitY/>
       <curve color="#f14cc1" name="/iq_model_calib_rt/iq_cal_node/tau_friction_meas.1"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="tab6" containers="1">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500561;0.499439">
     <DockSplitter count="2" orientation="|" sizes="0.5;0.5">
      <DockArea name="...">
       <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
        <range left="13.650000" top="0.131352" right="33.646000" bottom="-0.131551"/>
        <limitY/>
        <curve color="#d62728" name="/iq_model_calib_rt/iq_est_node/q_p_ddot_est_filt.0"/>
        <curve color="#1ac938" name="/iq_model_calib_rt/iq_est_node/q_p_ddot_est_filt.1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
        <range left="13.650000" top="0.004053" right="33.646000" bottom="-0.000095"/>
        <limitY/>
        <curve color="#bcbd22" name="/iq_model_calib_rt/iq_cal_node/q_dot_filt.0"/>
        <curve color="#1f77b4" name="/iq_model_calib_rt/iq_cal_node/q_dot_filt.1"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="2" orientation="|" sizes="0.5;0.5">
      <DockArea name="...">
       <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
        <range left="13.650000" top="0.100000" right="33.646000" bottom="-0.100000"/>
        <limitY/>
        <curve color="#9467bd" name="/iq_model_calib_rt/iq_cal_node/iq_meas.0"/>
        <curve color="#17becf" name="/iq_model_calib_rt/iq_cal_node/iq_meas.1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_x="false" style="Lines" mode="TimeSeries" flip_y="false">
        <range left="13.650000" top="-0.042288" right="33.646000" bottom="-0.093278"/>
        <limitY/>
        <curve color="#d62728" name="/iq_model_calib_rt/iq_est_node/tau_meas.0"/>
        <curve color="#1ac938" name="/iq_model_calib_rt/iq_est_node/tau_meas.1"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="8"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default delimiter="0" time_axis=""/>
  </plugin>
  <plugin ID="DataLoad MCAP"/>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;--]]&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. --]]&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value &#xa;  &#xa;--]]&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

